### About the Notebook

This notebook sets up the initial data ingestion by initialising all the relevant packages, reading source data from the Raw data store and converting it into a Dataframe for further analysis downstream. 
The data that is being ingested is `Grade_Profile_Mar23__Feb24.xlsx`

In [1]:
import pandas as pd

import sys
from pathlib import Path 

project_root = Path("..").resolve()
sys.path.append(str(project_root))

print(f"Project root added to path: {project_root}")

from src.utils import validate_schema, EXPECTED_COLUMNS

Project root added to path: C:\Users\ashma\projects\grade-profile-ml


In [2]:
raw_data_path = Path("../data/raw/Grade Profile Mar23 - Feb24.xlsx")

df_raw = pd.read_excel(raw_data_path)

In [3]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 335 entries, 0 to 334
Data columns (total 6 columns):
 #   Column                                Non-Null Count  Dtype         
---  ------                                --------------  -----         
 0   Current Programme (Aptem)             335 non-null    object        
 1   Apprenticeship Level                  335 non-null    int64         
 2   Achievement Date (ILR)                335 non-null    datetime64[ns]
 3   EPAO                                  335 non-null    object        
 4   aptem__UserILRSummary_GradingOutcome  318 non-null    object        
 5   Age Bracket                           329 non-null    object        
dtypes: datetime64[ns](1), int64(1), object(4)
memory usage: 15.8+ KB


## Input Data Summary
The data set has **335** row with **6** columns. The date variable was accurately parsed as a datetime variable and can be used to define end dates for different cohorts.
Apprenticeship level is an integer but I will convert this to a categorical variable as it is a dimension and not a measure. There are 2 columns with null values in it - `aptem__UserILRSummary_GradingOutcome`   and `Age Bracket`. I will mostly remove these rows as the main aim of this is to predict `aptem__UserILRSummary_GradingOutcome` column. 
However I will first check whether the nulls are randomly distributed or concentrated in a particular Level or EPAO, as non-random nulls could introduce bias into the model.
| Column | Null Count | Planned Action |
|---|---|---|
| `aptem__UserILRSummary_GradingOutcome` | 17 | Remove rows after null distribution check |
| `Age Bracket` | 6 | Remove rows after null distribution check |

In [4]:
# Validate input data

validate_schema(df_raw)

In [5]:
# Write to Parquet file

processed_data_path = Path("../data/processed/01_raw_loaded.parquet")
df_raw.to_parquet(processed_data_path,index=False)